hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -1000,7 +1000,9 @@ private static List<FileSlice> getPartitionFileSlices(HoodieTableMetaClient meta
     if (mergeFileSlices) {
       if (metaClient.getActiveTimeline().filterCompletedInstants().lastInstant().isPresent()) {
         fileSliceStream = fsView.getLatestMergedFileSlicesBeforeOrOn(
-            partition, metaClient.getActiveTimeline().filterCompletedInstants().lastInstant().get().getTimestamp());
+            // including pending compaction instant as the last instant so that the finished delta commits
+            // that start earlier than the compaction can be queried.
+            partition, metaClient.getActiveTimeline().filterCompletedAndCompactionInstants().lastInstant().get().getTimestamp());
```

Comment:
you mean to filter **completed** compaction instants and not the **pending** right? Comment above says pending compaction.

Reply:
yeah, just additionally including the pending compaction instants because now we do file slicing based on completion time, we should use compaction instant to filter out the file slices if there are del_commit with smaller start time but greater completion time (than the compaction)